pkg/config/validator.go

```diff
@@ -1,13 +1,32 @@
 package config
 
-import "reflect"
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/Axway/agent-sdk/pkg/util/log"
+)
 
 // ValidateConfig - Validates the agent config
 // Uses reflection to get the IConfigValidator interface on the config struct or
 // struct variable.
 // Makes call to ValidateCfg method except if the struct variable is of CentralConfig type
 // as the validation for CentralConfig is already done during parseCentralConfig
 func ValidateConfig(cfg interface{}) error {
+	obj := cfg
+	// This defer func is to catch a possible panic that WILL occur if the cfg object that is passed in embedds the IConfigValidator interface
+	// within its struct, but does NOT implement the ValidateCfg method. While it might be that this method really isn't necessary, we will
+	// log an error alerting the user in case it wasn't intentional.
+	defer func() {
```

Comment:
Can these defer funcs be pulled out into a sep func in the package, they are basically dupes of each other

Reply:
Yeah. I thought about doing that because Sonar is complaining about the complexity being a little high.. Will fix. Just need to pass the proper msg for each.

Reply:
done